internal/filesystem.go

```diff
@@ -72,9 +72,40 @@ func SystemDListener() (net.Listener, error) {
 	return net.FileListener(file)
 }
 
+type runDirListener struct {
+	listener   net.Listener
+	dirCreated bool
+}
+
+func (rdl *runDirListener) Accept() (net.Conn, error) {
+	return rdl.listener.Accept()
+}
+
+func (rdl *runDirListener) Close() error {
+	var removeErr error
```

Comment:
shouldn't also close also the `listener` here?

Reply:
It is closed in the return line